File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        longest_sequence = 0
        set_of_nums = set(nums)
        for num in nums:
            if num - 1 not in set_of_nums:
                length = 1
                while num + length in set_of_nums:
                    length += 1
                longest_sequence = max(longest_sequence, length)
        return longest_sequence

        

File: Data Structures & Algorithms/two-integer-sum/submission-2.cpp
class Solution {
public:
    vector<int> twoSum(vector<int>& nums, int target) {
        std::unordered_map<int, int> hashmap;
        for (int i = 0; i < nums.size(); i++){
            int num = nums[i];
            int need = target - num;
            if (hashmap.contains(need)){
                return {hashmap.at(need), i};
            }
            else{
                hashmap[num] = i;
            }
        }
        
    }
};
